wfe/web-front-end.go

```diff
@@ -1033,17 +1028,34 @@ func (wfe *WebFrontEndImpl) Certificate(response http.ResponseWriter, request *h
 		return
 	}
 
-	addCacheHeader(response, wfe.CertCacheDuration.Seconds())
-
-	// TODO Content negotiation
-	response.Header().Set("Content-Type", "application/pkix-cert")
-	response.Header().Add("Link", link(IssuerPath, "up"))
-	response.WriteHeader(http.StatusOK)
-	if _, err = response.Write(cert.DER); err != nil {
-		logEvent.Error = err.Error()
-		wfe.log.Warning(fmt.Sprintf("Could not write response: %s", err))
+	// If the signing request is still pending, just return a 202 Accepted
+	switch cert.Status {
+	case core.StatusPending:
+		// Return a polite 202
+		addNoCacheHeader(response)
+		response.Header().Add("Link", link(IssuerPath, "up"))
+		response.WriteHeader(http.StatusAccepted)
+	case core.StatusInvalid:
+		// Return 410 Gone
+		addNoCacheHeader(response)
```

Comment:
It's not like this 410 is going to change to a nicer response.  Shouldn't we allow caching?